homeassistant/components/gocr/image_processing.py

```diff
@@ -0,0 +1,162 @@
+    vol.Optional(CONF_UNRECOGNIZED): cv.string,
+    vol.Optional(CONF_CHARS): cv.string,
+    vol.Inclusive(CONF_HEIGHT, _KEY_GEOMETRY): cv.positive_int,
+    vol.Inclusive(CONF_X_POSITION, _KEY_GEOMETRY): cv.positive_int,
+    vol.Inclusive(CONF_WIDTH, _KEY_GEOMETRY): cv.positive_int,
+    vol.Inclusive(CONF_Y_POSITION, _KEY_GEOMETRY): cv.positive_int,
+    vol.Optional(CONF_GOCR_BIN, default=DEFAULT_BINARY): cv.string,
+    vol.Optional(CONF_THRESHOLD, default=50): cv.positive_int,
+    vol.Optional(CONF_ROTATE, default=0): cv.positive_int,
+
+    vol.Optional(CONF_NEGATE, default=False): cv.boolean,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the GOCR platform."""
+    entities = []
+    for camera in config[CONF_SOURCE]:
+        entities.append(ImageProcessingGocr(
+            hass, camera[CONF_ENTITY_ID], config, camera.get(CONF_NAME)
+        ))
+
+    async_add_entities(entities)
+
+
+class ImageProcessingGocr(ImageProcessingEntity):
+    """Representation of GOCR image processing entity."""
+
+    def __init__(self, hass, camera_entity, config, name):
+        """Initialize text processing."""
+        self._camera_entity = camera_entity
+        if name is not None:
+            self._name = name
+        else:
+            self._name = ("GOCR {0}".format(
+                split_entity_id(camera_entity)[1])
+                if config[CONF_NAME] == '' else
+                config[CONF_NAME])
+        self._state = None
+        if hasattr('config', CONF_X_POSITION) and config[CONF_X_POSITION] is not None:
+            self.crop = (config[CONF_X_POSITION], config[CONF_Y_POSITION],
+                         config[CONF_X_POSITION] + config[CONF_WIDTH],
+                         config[CONF_Y_POSITION] + config[CONF_HEIGHT])
+        else:
+            self.crop = None
+        self.rotate = config[CONF_ROTATE]
+        self.negate = config[CONF_NEGATE]
+        if config.get(CONF_CHARS) is not None:
+            digits = ['-C', config.get(CONF_CHARS)]
+        else:
+            digits = []
+        if config.get(CONF_UNRECOGNIZED) is not None:
+            unrecognized = ['-u', config.get(CONF_UNRECOGNIZED)]
+        else:
+            unrecognized = []
+        if config[CONF_THRESHOLD] is not None:
+            threshold = ['-l', str(config[CONF_THRESHOLD])]
+        else:
+            threshold = []
+        if config[CONF_EXTRA_ARGUMENTS] is not None:
+            extra_arguments = shlex.split(config[CONF_EXTRA_ARGUMENTS])
+        else:
+            extra_arguments = []
+
+        self._command = ([config[CONF_GOCR_BIN],
+                         '-e', '/dev/null', '-f', 'UTF8'] +
+                         digits + unrecognized + threshold +
+                         extra_arguments)
+        _LOGGER.info("Command : %s -i <tmpfile>",
+                     shlex.quote(' '.join(self._command)))
```

Comment:
```suggestion
                     ' '.join(shlex.quote(x) for x in self._command))
```

Technically, this is the actual command run - not the other way around with shlex.quote